src/MergeContext.js

```diff
@@ -454,6 +454,202 @@ class BranchPosition
+        this._attributes = ["Authored-by", "Co-authored-by"];
+
+        if (rawPr.body !== undefined && rawPr.body !== null) {
+            this._body = rawPr.body.replace(/\r+\n/g, '\n').trimEnd();
+            this._parse();
+        }
+    }
+
+    // parsing error details
+    errorMessage() {
+        assert(!this.valid());
+        return this._error.message;
+    }
+
+    // whether the raw PR message has been parsed successfully
+    valid() { return !this._error; }
+
+    // complete message for the staged commit
+    whole() {
+        assert(this.valid());
+        assert(this._author !== null && this._body !== null && this._trailer !== null);
+        let message = this._title;
+        if (this._body)
+            message += '\n\n' + this._body;
+        if (this._trailer)
+            message += '\n\n' + this._trailer;
+        return message;
+    }
+
+    // returns author object in the {name, email, date} format
+    createAuthor(mergeCommitAuthor)
+    {
+        if (!this._author)
+            return mergeCommitAuthor;
+        return {name: this._author.name, email: this._author.email, date: mergeCommitAuthor.date};
+    }
+
+    author() { return this._author; }
+
+    // returns the position of the first non-ASCII_printable character (or -1)
+    _invalidCharacterPosition(str) {
+        const prohibitedCharacters = /[^\u{20}-\u{7e}]/u;
+        const match = prohibitedCharacters.exec(str);
+        return match ? match.index : -1;
+    }
+
+    // basic checks for the entire raw message
+    _checkMessage() {
+        // constructor removed CRs in CRLF sequences
+        // other CRs are not treated specially (and are banned)
+        const lines = this.whole().split('\n');
+        for (let i = 0; i < lines.length; ++i) {
+            const line = lines[i];
+            const invalidPosition = this._invalidCharacterPosition(line);
+
+            if (invalidPosition !== -1)
+                throw new Error(`bad character at line ${i}, offset ${invalidPosition}`);
+
+            if (line.length > 72)
+                throw new Error(`too long line '${line}'`);
+        }
+    }
+
+    // checks whether the message has a misused attribute
+    _checkAttributes() {
+        // A line matched by this expression is considered valid if it
+        // starts with one of the allowed attributes and invalid otherwise.
+        // A line that is not matched by this expression and containing one
+        // of the allowed attributes is ignored.
+        const authorExp = /^\s*\S*[aA]uthored[-_]?[bB]y/;
+        let lines = this._body.split('\n');
+        while (lines.length > 0) {
+            const line = lines.shift();
+            if (line.match(authorExp)) {
+                if (this._attributes.some(attr => line.startsWith(attr + ':')))
+                    continue;
+                throw new Error(`a misused '*Authored-by' attribute in the message`);
+            }
+        }
+    }
```

Comment:
Perhaps I am missing something, but I see no reason to add checkAttributes() and a fixed list of field names like this._attributes: Each loop iterating nextField()s already knows which fields are acceptable in _that_ loop context (header or trailer) and is in a much better position to handle things like unknown fields (that we want to allow in the trailer). What will change/break if we remove all the functionality associated with this._attributes? If nothing, let's remove this method and related code.